src/core/server/opensearch/version_check/ensure_opensearch_version.ts

```diff
@@ -36,16 +36,44 @@
  */
 
 import { timer, of, from, Observable } from 'rxjs';
-import { map, distinctUntilChanged, catchError, exhaustMap } from 'rxjs/operators';
+import { map, distinctUntilChanged, catchError, exhaustMap, mergeMap } from 'rxjs/operators';
+import { get, forEach } from 'lodash';
 import {
   opensearchVersionCompatibleWithOpenSearchDashboards,
   opensearchVersionEqualsOpenSearchDashboards,
 } from './opensearch_opensearch_dashboards_version_compatability';
 import { Logger } from '../../logging';
 import type { OpenSearchClient } from '../client';
 
+export const getNodeId = async (internalClient: OpenSearchClient): Promise<string | null> => {
+  try {
+    const state = await internalClient.cluster.state({
+      filter_path: ['nodes.*.attributes.cluster_id'],
+    });
+    // Aggregate different cluster_ids from the OpenSearch nodes
+    const clusterIdSet = new Set();
+    let defaultClusterId = 0;
+
+    // if attributes.cluster_id is missing, assign the monotonically increasing default cluster_id for each node
+    forEach(state.body.nodes, (node: any) => {
+      clusterIdSet.add(get(node, 'attributes.cluster_id', defaultClusterId++));
+    });
+
+    /* if all the nodes have the same cluster_id, retrieve nodes.info from _local node only
+     * Using _cluster/state/nodes to retrieve the cluster_id of each node from master node which is considered to be a lightweight operation
+     * else if the nodes have different cluster_ids then fan out the request to all nodes
+     * else there are no nodes in the cluster
+     */
+    const nodeId = clusterIdSet.size === 1 ? '_local' : null;
+    return nodeId;
```

Comment:
Not entirely relevant if you change the logic based on my recommendation above, but can't we return the result instead of creating the variable?